src/modules/silentpayments/main_impl.h

```diff
@@ -7,10 +7,313 @@
+     * Returning an error here results in an untestable branch in the code, but we do this anyways to ensure strict compliance with BIP0352.
+     */
+    if (!secp256k1_silentpayments_create_output_tweak(&output_tweak_scalar, shared_secret33, k)) {
+        return 0;
+    }
+    if (!secp256k1_pubkey_load(ctx, &output_ge, spend_pubkey)) {
+        secp256k1_scalar_clear(&output_tweak_scalar);
+        return 0;
+    }
+    /* `tweak_add` only fails if output_tweak_scalar*G = -spend_pubkey. Considering output_tweak is the output of a hash function,
+     * this will happen only with negligible probability for honestly created spend_pubkey, but we handle this
+     * error anyway to protect against this function being called with a malicious inputs, i.e., spend_pubkey = -(_create_output_tweak(shared_secret33, k))*G
+     */
+    if (!secp256k1_eckey_pubkey_tweak_add(&output_ge, &output_tweak_scalar)) {
+        secp256k1_scalar_clear(&output_tweak_scalar);
+        return 0;
+    };
+    secp256k1_xonly_pubkey_save(output_xonly, &output_ge);
+
+    /* Leaking this value would break indistinguishability of the transaction, so clear it. */
+    secp256k1_scalar_clear(&output_tweak_scalar);
+    return 1;
+}
+
+int secp256k1_silentpayments_sender_create_outputs(
+    const secp256k1_context *ctx,
+    secp256k1_xonly_pubkey **generated_outputs,
+    const secp256k1_silentpayments_recipient **recipients,
+    size_t n_recipients,
+    const unsigned char *outpoint_smallest36,
+    const secp256k1_keypair * const *taproot_seckeys,
+    size_t n_taproot_seckeys,
+    const unsigned char * const *plain_seckeys,
+    size_t n_plain_seckeys
+) {
+    size_t i, k;
+    secp256k1_scalar seckey_sum_scalar, addend, input_hash_scalar;
+    secp256k1_ge prevouts_pubkey_sum_ge;
+    secp256k1_gej prevouts_pubkey_sum_gej;
+    unsigned char shared_secret[33];
+    secp256k1_pubkey current_scan_pubkey;
+    int ret, sum_is_zero;
+
+    /* Sanity check inputs. */
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secp256k1_ecmult_gen_context_is_built(&ctx->ecmult_gen_ctx));
+    ARG_CHECK(generated_outputs != NULL);
+    ARG_CHECK(recipients != NULL);
+    ARG_CHECK(n_recipients > 0);
+    ARG_CHECK(outpoint_smallest36 != NULL);
+    ARG_CHECK((plain_seckeys != NULL) || (taproot_seckeys != NULL));
+    if (taproot_seckeys != NULL) {
+        ARG_CHECK(n_taproot_seckeys > 0);
+    } else {
+        ARG_CHECK(n_taproot_seckeys == 0);
+    }
+    if (plain_seckeys != NULL) {
+        ARG_CHECK(n_plain_seckeys > 0);
+    } else {
+        ARG_CHECK(n_plain_seckeys == 0);
+    }
+    for (i = 0; i < n_recipients; i++) {
+        ARG_CHECK(recipients[i]->index == i);
+    }
+
+    seckey_sum_scalar = secp256k1_scalar_zero;
+    for (i = 0; i < n_plain_seckeys; i++) {
+        ret = secp256k1_scalar_set_b32_seckey(&addend, plain_seckeys[i]);
```

Comment:
nit: should check/assert `plain_seckeys[I]` is not NULL. Otherwise this will crash.

Reply:
Good catch, will add an `ARG_CHECK` to check for non-NULL, here and in other API functions (scanning) where similar checks for "array of pointers" elements are missing. I've also opened https://github.com/bitcoin-core/secp256k1/pull/1779 to do the same for API functions in master for consistency.